docker/single-process/README.md

```diff
@@ -0,0 +1,94 @@
+Docker image for Huginn using the production environment and separate container for every process
+=================================================
+
+This image runs a linkable [Huginn](https://github.com/cantino/huginn) instance.
+
+It was inspired by the [official docker container for huginn](https://registry.hub.docker.com/u/cantino/huginn)
+
+The scripts/init script generates a .env file containing the variables as passed as per normal Huginn documentation.
+The same environment variables that would be used for Heroku PaaS deployment are used by this script.
+
+The scripts/init script is aware of mysql and postgres linked containers through the environment variables:
+
+    MYSQL_PORT_3306_TCP_ADDR
+    MYSQL_PORT_3306_TCP_PORT
+
+and
+
+    POSTGRESQL_PORT_5432_TCP_ADDR
+    POSTGRESQL_PORT_5432_TCP_PORT
+
+Its recommended to use an image that allows you to create a database via environmental variables at docker run, like `postgresql` or `mysql`, so the db is populated when this script runs.
+
+Additionally, the database variables may be overridden from the above as per the standard Huginn documentation:
+
+    DATABASE_ADAPTER #(must be either 'postgresql' or 'mysql2')
+    DATABASE_HOST
+    DATABASE_PORT
+
+This script will run database migrations (rake db:migrate) which should be idempotent.
+
+It will also seed the database (rake db:seed) unless this is defined:
+
+    DO_NOT_SEED
+
+This same seeding initially defines the "admin" user with a default password of "password" as per the standard Huginn documentation.
+
+If you do not wish to have the default 6 agents, you will want to set the above environment variable after your initially deploy, otherwise they will be added automatically the next time a container pointing at the database is spun up.
+
+The CMD launches Huginn via the scripts/init script. This may become the ENTRYPOINT later.  It does take under a minute for Huginn to come up.  Use environmental variables that match your DB's creds to ensure it works.
+
+## Usage
+
+Simple startup using docker compose (you need to daemonize with `-d` to persist the data):
+
+    cd docker/single-process
+    docker-compose up
```

Comment:
Toward the end of doing a local `docker-compose up`, there were a few database connection errors, but it does look like Huginn is running:

```
huginn_web_1      | Migrating to WebsiteAgentDoesNotUseEventUrl (20160108221620)
huginn_web_1      | == 20160108221620 WebsiteAgentDoesNotUseEventUrl: migrating ===================
huginn_web_1      | == 20160108221620 WebsiteAgentDoesNotUseEventUrl: migrated (0.0052s) ==========
huginn_web_1      |
mysql_1           | 2016-01-24T04:37:46.515426Z 6 [Note] Aborted connection 6 to db: 'huginn' user: 'huginn' host: '172.17.0.3' (Got an error reading communication packets)
huginn_web_1      |
huginn_web_1      |
huginn_web_1      | NOTE: The example 'SF Weather Agent' will not work until you edit it and put in a free API key from http://www.wunderground.com/weather/api/
huginn_web_1      | See the Huginn Wiki for more Agent examples!  https://github.com/cantino/huginn/wiki
mysql_1           | 2016-01-24T04:37:53.370271Z 9 [Note] Aborted connection 9 to db: 'huginn' user: 'huginn' host: '172.17.0.3' (Got an error reading communication packets)
huginn_web_1      | I, [2016-01-24T04:37:54.695421 #720]  INFO -- : Refreshing Gem list
huginn_web_1      | I, [2016-01-24T04:38:00.147728 #720]  INFO -- : listening on addr=0.0.0.0:3000 fd=8
mysql_1           | 2016-01-24T04:38:00.154671Z 11 [Note] Aborted connection 11 to db: 'huginn' user: 'huginn' host: '172.17.0.3' (Got an error reading communication packets)
huginn_web_1      | I, [2016-01-24T04:38:00.160977 #720]  INFO -- : master process ready
huginn_web_1      | I, [2016-01-24T04:38:00.166350 #742]  INFO -- : worker=1 ready
huginn_web_1      | I, [2016-01-24T04:38:00.170625 #739]  INFO -- : worker=0 ready
```